### PR TITLE
Bwhite merge master into 8070

### PR DESCRIPTION
**Pull request by BillWhite**
There are 228 test failures in this branch.  There are 240 in the previous branch.  The
differences in community test failures are below.  Note that all are in the test suite
TestSubQueriesSuite.

Failures in bwhite-MergeMasterInto8070 not in ENG-8070-hsql232-upgrade.
  org.voltdb.TestJSONInterface.testDeployment (manual pass)
  org.voltdb.regressionsuites.TestSubQueriesSuite.testEng8394SubqueryWithUnionAndCorrelation-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testEng8394SubqueryWithUnionAndCorrelation-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testFromClauseAggregation-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testFromClauseAggregation-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testFromReplicated-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testFromReplicated-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testInExistsOrOpAnyWithInnerNull-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testJoinsOfInExists-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testJoinsOfSubselects-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testJoinsOfSubselects-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testLhsScalarInSubquery-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testLhsScalarInSubquery-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testOpAllWithInnerNull-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testOpAllWithInnerNull-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testOpAllWithOuterNull-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testOpAllWithOuterNull-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testOuterNullInOpAny-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testOuterNullInOpAny-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testRowEqualityIsNull-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testRowEqualityIsNull-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSelectScalar-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSelectScalar-localCluster-2-1-JNI

Failures in ENG-8070-hsql232-upgrade not in bwhite-MergedMasterInto8070
  org.voltdb.regressionsuites.TestSubQueriesSuite.testALLSubExpressions_InnerNull-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testALLSubExpressions_InnerNull-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testALLSubExpressions_InnerNull-localCluster-2-3-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testALLSubExpressions_OuterNull-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testALLSubExpressions_OuterNull-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testALLSubExpressions_OuterNull-localCluster-2-3-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testANYSubExpressions_OuterNull-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testANYSubExpressions_OuterNull-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testANYSubExpressions_OuterNull-localCluster-2-3-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testENG6276-localCluster-2-3-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testExists_Joins-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testExists_Joins-localCluster-2-3-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testRepeatedQueriesDifferentData-localCluster-2-3-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSelectScalarSubSelects-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSelectScalarSubSelects-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSelectScalarSubSelects-localCluster-2-3-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubExpressions_InnerNull-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubExpressions_InnerNull-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubExpressions_InnerNull-localCluster-2-3-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubExpressions_Simple-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubExpressions_Simple-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubExpressions_Simple-localCluster-2-3-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubSelects_Aggregations-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubSelects_Aggregations-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubSelects_Aggregations-localCluster-2-3-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubSelects_Joins-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubSelects_Joins-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubSelects_Joins-localCluster-2-3-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubSelects_from_replicated-localCluster-1-1-HSQLDB
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubSelects_from_replicated-localCluster-2-1-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubSelects_from_replicated-localCluster-2-3-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubqueriesWithArithmetic-localCluster-2-3-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testSubqueryWithExceptions-localCluster-2-3-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testWhereRowSubSelects-localCluster-2-3-JNI
  org.voltdb.regressionsuites.TestSubQueriesSuite.testWhereScalarSubSelects-localCluster-2-3-JNI

### CODE REVIEW

**Comment by wolffcm** [comment]
I guess some tests started passing as a result of changes to master?  I wonder why that is.  

Obviously we can't review all the changes here.  Can you point me to places where you made manual changes, or that you feel need attention, and I can take a look?

File: src/frontend/org/voltdb/dbmonitor/js/voltdb.core.js [hunk 1]
```diff
@@ -219,8 +224,8 @@
                 return this;
             };
 
-            this.BeginExecute = function (procedure, parameters, callback, shortApiCallDetails) {
-                var isHighTimeout = (procedure == "@SnapshotRestore" || procedure == "@AdHoc");
+            this.BeginExecute = function (procedure, parameters, callback, shortApiCallDetails, isLongOutput) {
```
Comment (BillWhite) [inline src/frontend/org/voltdb/dbmonitor/js/voltdb.core.js]: I wasn't sure what was the right thing to do here.

File: tests/frontend/org/voltdb/regressionsuites/TestSystemProcedureSuite.java [hunk 2]
```diff
@@ -396,6 +397,87 @@ public void testProfCtl() throws Exception {
         assertTrue(true);
     }
 
+    public void testPause() throws Exception {
```
Comment (BillWhite) [inline tests/frontend/org/voltdb/regressionsuites/TestSystemProcedureSuite.java]: This test was very confusing.  I think I figured out what was going on, but I'm not sure.

**Comment by BillWhite** [comment]
I have marked the few places I found confusing in the pull request.

I'll look into it tomorrow, by repeating the merge on a new branch, and
then seeing where I was confused.
I feel there was something else, but I don't see it now.

File: src/hsqldb19b3/org/hsqldb_voltpatches/FunctionSQL.java [hunk 3]
```diff
@@ -398,9 +398,6 @@ protected FunctionSQL(int id) {
             case FUNC_MOD :
```
Comment (BillWhite) [inline src/hsqldb19b3/org/hsqldb_voltpatches/FunctionSQL.java]: I'm not sure I got all these voltDisabled cases right.

**Pull request merged**